src/systems/hubs-systems.js

```diff
@@ -23,16 +26,18 @@ AFRAME.registerSystem("hubs-systems", {
     this.hapticFeedbackSystem = new HapticFeedbackSystem();
     this.soundEffectsSystem = new SoundEffectsSystem();
     this.lobbyCameraSystem = new LobbyCameraSystem();
+    this.spriteSystem = new SpriteSystem(this.el);
     this.batchManagerSystem = new BatchManagerSystem(this.el.sceneEl.object3D, this.el.sceneEl.renderer);
   },
 
-  tick(t) {
+  tick(t, dt) {
     const systems = AFRAME.scenes[0].systems;
     systems.userinput.tick2();
-    systems.interaction.tick2(this.soundEffectsSystem);
     this.superSpawnerSystem.tick();
     this.cursorTargettingSystem.tick(t);
     this.constraintsSystem.tick();
+    this.physicsSystem.tick(dt);
+    systems.interaction.tick2(this.soundEffectsSystem);
```

Comment:
Not sure what the optimal tick order is (physics before interaction or interaction before physics), but since `superSpawnerSystem`, `cursorTargettingSystem` and `constraintsSystem` read from `interaction` system state, I think they should come after the `interaction` system tick. In the current order, those three systems have the previous frame's interaction system state but the current frame's userinput state. I suggest `tick`ing physics either first or after all interactions have happened, unless there's a reason I'm missing for why we want it to happen in the middle.

Reply:
I didn't know those systems read from `interaction` system, just wanted to make sure `interaction` ticked after physics did. In that case, I agree, `physics` tick should just happen first.

Reply:
Oh except `constraintSystem` needs to tick before `physicsSystem`